src/main/java/seedu/duke/parser/ParameterParser.java

```diff
@@ -413,10 +423,8 @@ public static String parseCategoryTag(String parameter) throws InputTransactionI
      * @throws InputTransactionInvalidAmountException If the transaction amount given is not a valid accepted integer.
      */
     private static int parseAmountTag(String parameter) throws InputTransactionInvalidAmountException {
-        Pattern specialSymbols = Pattern.compile("[!@#$%&*()_+=|<>?{}\\[\\]~-]");
-        Matcher hasSpecialSymbols = specialSymbols.matcher(parameter);
         try {
-            if (containAlphabet(parameter) || hasSpecialSymbols.find()) {
+            if (containAlphabet(parameter) || containSymbol(parameter)) {
```

Comment:
Great attempt at replacing the default  `hasSpecialSymbols.find()` Java function with a custom handler.